flux-tests/tests/pos/enums/list01.rs

```diff
@@ -0,0 +1,80 @@
+#![feature(register_tool)]
+#![register_tool(flux)]
+#![feature(custom_inner_attributes)]
+#![flux::defs {
+    fn set_add(x: int, s: Set<int>) -> Set<int> { set_union(set_singleton(x), s) }
+    fn set_is_empty(s: Set<int>) -> bool { s == set_empty(0) }
+    fn set_emp() -> Set<int> { set_empty(0) }
+}]
+
+/// (i32) lists indexed by the _set_ of elements.
+
+#[flux::refined_by(elems: Set<int>)]
+pub enum List {
+    #[flux::variant(List[set_emp()])]
+    Nil,
+    #[flux::variant((i32[@n], Box<List[@elems]>) -> List[set_add(n, elems)])]
+    Cons(i32, Box<List>),
+}
+
+#[flux::sig(fn(&List[@xs]) -> bool[xs == set_emp()])]
+pub fn is_empty(l: &List) -> bool {
+    match l {
+        List::Nil => true,
+        List::Cons(_, _) => false,
+    }
+}
+
+// WHY DOES THIS FAIL?
+#[flux::sig(fn () -> List{v:v==set_emp()})]
+
+// #[flux::sig(fn () -> List[set_emp()])]
+pub fn null() -> List {
+    List::Nil
+}
+
+#[flux::sig(fn(i32{v: false}) -> T)]
+pub fn never<T>(_: i32) -> T {
+    loop {}
+}
```

Comment:
I typically call this `unreachable` like the rust macro `unreachable!` and define it like so

```suggestion
#[flux::sig(fn() -> T requires false)]
pub fn unreachable<T>() -> T {
    unreachable!()
}
```